defaults/main.yml

```diff
@@ -8,10 +8,13 @@ bind9_group: bind
 bind9_ipv6: yes
 
 # Run bind as a DNS recursor?
+# variable used only by default templates, not strict_authoritative
 bind9_recursor: no
 
 # Run bind as authoritative nameserver?
-bind9_authoritative: no
+# variable by default templates and as conditionnal of several tasks
+# If using `strict_authoritative/` templates, this variable _must_ be true
+bind9_authoritative: "{{ true if bind9_templates == 'strict_authoritative/' else false }}"
```

Comment:
I think yes or no would be better the better choice here since all other variables are also yes or no.

Reply:
With this conditional default definition I want to achieve two goals: 
* preserve the default value `no` (or `false`, better to avoid warning in lint) for the default templates, avoiding breaking behavior in other deployments of the role, 
* set the value to `yes` when selecting `strict_authoritative` templates, as it is always needed for the role's tasks logic to configure zones.